### PR TITLE
Hyperion 1036 Set energy: integrate set energy functions into hyperion plans

### PR DESCRIPTION
**Pull request by rtuck99**
Fixes DiamondLightSource/hyperion#1036

Refactors the DCM fixed offset so that now dodal populates an attribute on the device instead of making hyperion parse the beamline parameters.

Adds undulatorDCM as part of beamline I03 devices as it was missing.

### Instructions to reviewer on how to test:
- [x] Tests pass

### Checks for reviewer
- [x] Would the PR title make sense to a scientist on a set of release notes
- [x] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

File: tests/beamlines/unit_tests/test_i03.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def test_device_creation():
     beamline_utils.clear_devices()
     devices = make_all_devices(i03, fake_with_ophyd_sim=True)
     for device_name in devices.keys():
+        print(device_name)
```
Comment (olliesilvester) [inline tests/beamlines/unit_tests/test_i03.py:22]: We can remove this now

File: src/dodal/beamlines/i03.py [hunk 2]
```diff
@@ -94,19 +95,6 @@ def vfm_mirror_voltages(
     )
 
 
-@skip_device(lambda: BL == "s03")
-def hfm(
```
Comment (olliesilvester) [inline src/dodal/beamlines/i03.py:98]: How come this device is removed now?

Comment (rtuck99) [reply src/dodal/beamlines/i03.py:98]: So initially when I first started working on the energy changes, I based it off of what GDA currently does, however after Neil put me right it turns out that we don't actually do any changes with it, everything is handled with feedback so no need to move it. Only the vertical mirror is needed.

**Review by olliesilvester — APPROVED** [review]
Looks good other than that print statement in the test

**Review by olliesilvester — APPROVED** [review]
Looks sensible, thanks!

**Pull request merged**